vispy/visuals/_scalable_textures.py

```diff
@@ -355,23 +388,22 @@ def _get_gl_tex_format(self, texture_format, num_channels):
         return texture_format
 
     def _get_texture_format_for_data(self, data, internalformat):
-        if internalformat is not None:
-            num_channels = self._data_num_channels(data)
-            texture_format = self._handle_auto_texture_format(internalformat, data)
-            texture_format = self._get_gl_tex_format(texture_format, num_channels)
+        assert internalformat is not None
```

Comment:
```suggestion
        if internalformat is None:
            raise ValueError("'internalformat' must be provided for GPU scaled textures.")
```

Some code quality services (we don't have any here right now) complain about using asserts outside of tests. Plus it might be nice to have an explicit error message.